sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/PropagatorFactory.java

```diff
@@ -27,9 +31,41 @@ static PropagatorFactory getInstance() {
   @Override
   public ContextPropagators create(
       PropagatorModel model, SpiHelper spiHelper, List<Closeable> closeables) {
-    List<String> compositeModel = requireNonNull(model.getComposite(), "composite propagator");
-    TextMapPropagator textMapPropagator =
-        TextMapPropagatorFactory.getInstance().create(compositeModel, spiHelper, closeables);
-    return ContextPropagators.create(textMapPropagator);
+    List<TextMapPropagatorModel> textMapPropagatorModels = model.getComposite();
+    Set<String> propagatorNames = new HashSet<>();
+    List<TextMapPropagator> textMapPropagators = new ArrayList<>();
+    if (textMapPropagatorModels != null && !textMapPropagatorModels.isEmpty()) {
```

Comment:
the `.forEach()` is effectively a no-op in the empty case, so that check can be safely removed.
```suggestion
    if (textMapPropagatorModels != null) {
```